content/post/elm-in-rakuten/index.md

```diff
@@ -0,0 +1,1273 @@
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+## 11. The Elm Architecture
+
+**The Elm Architecture** is probably the most relevant and influential innovation of Elm.[^the-elm-architecture] It is a **unidirectional data flow**[^unidirectional-data-flow] that helps to keep your application well organized. Also, it helps you to quickly understand applications built by other developers as this is the standard way to build applications in Elm.    
+    
+[^the-elm-architecture]: Details about The Elm Architecture can be found in the [official Elm Guide](https://guide.elm-lang.org/architecture/). The Elm Architecture is the predominant way to build applications in Elm. Different variants are also possible. [Elm-spa](https://www.elm-spa.dev/guide) by Ryan Haskell-Glatz is a tool that helps to create single-page applications and create extra abstraction above The Elm Architecture. Rémi Lefèvre built the [RealWorld example app](https://github.com/gothinkster/realworld) using the [Effect pattern](https://discourse.elm-lang.org/t/realworld-example-app-architected-with-the-effect-pattern/5753).
+        
+[^unidirectional-data-flow]: The Elm Architecture is based on [**unidirectional data flow**](https://youtu.be/jl1tGiUiTtI?t=470) (a.k.a. **one-way data binding**) like React, in contrast to the **bidirectional data flow** (a.k.a. **two-way data binding**) of frameworks like Angular, Vue, and Svelte ([in Svelte two-way binding can be disabled](https://github.com/sveltejs/svelte/issues/54)). There have been issues with two-way data binding. For example, the many-to-many dependencies between the view and the model can create an infinite loop of cascading updates. Another issue is the lack of control of the change detection mechanism. It is an implicit behavior that is not easy to control. Unidirectional data flow tends to be more predictable.
+    
+![The Elm Architecture, animated](https://lucamug.github.io/post-Elm-in-Rakuten/images/the-elm-architecture.svg)    
+*A simple representation of the unidirectional data flows in The Elm Architecture. (Source: The Elm Guide)*[^representation-of-unidirectional-flow]
+    
+[^representation-of-unidirectional-flow]: The illustration **A simple representation of the Elm Architecture** is from the as depicted in the [Elm Guide](https://guide.elm-lang.org/architecture/) used under the [CC BY-NC-ND 4.0](https://creativecommons.org/licenses/by-nc-nd/4.0/) [license](https://github.com/evancz/guide.elm-lang.org/blob/master/LICENSE).
+    
+There are three building blocks in The Elm Architecture:
+    
+* `Model` --- the state of your application, the only thing that can mutate
+* `view` --- a way to turn your state into HTML
+* `update` --- a way to update your state based on the `Model` and the *messages*
+        
+If we zoom in on the `Elm` block in the diagram above, this is what is happening inside:        
+
+![The Elm Architecture, animated](https://lucamug.github.io/post-Elm-in-Rakuten/images/the-elm-architecture-animation.gif)
```

Comment:
The image should be hosted here IMHO.
Also, it seems the same image is used multiple times, in different formats? (gif, png, svg…)